### PR TITLE
reindex-job should not mint doi when the work is private or under review

### PR DESCRIPTION
**Pull request by mankind**
This refactor prevents the ReindexModelJob from minting doi when a work is private or when the work is under review.

### CODE REVIEW

File: spec/jobs/reindex_model_job_spec.rb [hunk 1]
```diff
@@ -24,11 +33,39 @@
...
+      described_class.perform_now(work.class.to_s, account.cname, limit: 1, options: options)
+      expect(work.reload.doi_status_when_public).to eq "findable"
+    end
   end
 
-  it "sets doi_status_when_public to findable" do
-    described_class.perform_now(work.class.to_s, account.cname, limit: 1, options: options)
-    expect(work.reload.doi_status_when_public).to eq "findable"
+  context "skip doi minting" do
+    let(:sipity_entity) { instance_double(Sipity::Entity, proxy_for_global_id: "gid://hyku/#{pending_review_work.class}/#{pending_review_work.id}", workflow_id: workflow.id, workflow_state: sipity_workflow_state) }
```
Comment (prdanelli) [inline spec/jobs/reindex_model_job_spec.rb:54]: Could we shorten these longer lines, they're getting kind of gnarly, this one and L44. 

```
let(:gid) { "gid://hyku/#{pending_review_work.class}/#{pending_review_work.id}" }
let(:sipity_entity) { instance_double(Sipity::Entity, proxy_for_global_id: gid, workflow_id: workflow.id, workflow_state: sipity_workflow_state) }
```

Comment (mankind) [reply spec/jobs/reindex_model_job_spec.rb:54]: Thanks I will :+1:

File: app/jobs/hyku_addons/reindex_model_job.rb [hunk 2]
```diff
@@ -35,14 +35,19 @@ def perform(klass, cname, limit: 25, page: 1, options: {})
     private
 
       def mint_doi(work)
-        return if work.doi.present?
+        return if work.doi.present? || work.visibility != "open" || workflow_state(work)&.workflow_state_name != "deposited"
```
Comment (prdanelli) [inline app/jobs/hyku_addons/reindex_model_job.rb:38]: I find that when the guard clause starts to have more than one or two clauses to it, then it is better to add it to its own method. With this example there are positive and negative checks too. I would use something like this: 

```
return unless can_mint_for?(work)

def can_mint_for?(work)
    work.doi.blank? || work.visibility == "open" || workflow_state(work)&.workflow_state_name == "deposited"
end
```

Comment (mankind) [reply app/jobs/hyku_addons/reindex_model_job.rb:38]: Thanks will update it

Comment (mankind) [reply app/jobs/hyku_addons/reindex_model_job.rb:38]: I made the change to use unless with these conditions:

    work.doi.blank? || work.visibility == "open" || workflow_state(work)&.workflow_state_name == "deposited"
    
but the test failed so I reverted it back to what it was  but extracted it into a method like you did.
I thought i should let you know

**Review by prdanelli — APPROVED** [review]

**Pull request merged**